membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -0,0 +1,50 @@
+package models
+import com.gu.membership.salesforce.{Member, Contact, MemberStatus, PaymentMethod}
+import com.gu.services.PaymentDetails
+import play.api.libs.json._
+import play.api.mvc.Results.Ok
+
+object AccountDetails {
+  implicit class ResultLike(details: (Contact[MemberStatus, PaymentMethod], PaymentDetails)) {
+
+    def toResult = {
+      val contact = details._1
+      val paymentDetails = details._2
+      Ok(basicDetails(contact) ++ memberDetails(contact) ++ toJson(paymentDetails))
+    }
+
+    private def memberDetails(contact: Contact[MemberStatus, PaymentMethod]): JsObject = contact.memberStatus match {
+      case m: Member => Json.obj("regNumber" -> m.regNumber.mkString, "tier" -> m.tier.name, "isPaidTier" -> m.tier.isPaid)
+      case _ => Json.obj()
+    }
+
+    private def basicDetails(contact: Contact[MemberStatus, PaymentMethod]) =
+      Json.obj("joinDate" -> contact.joinDate)
+
+    private def toJson(paymentDetails: PaymentDetails): JsObject = {
```

Comment:
So this was taken from membership-frontend and since the JSON format is a bit odd and specific to the MMA page I didn't think it'd be worth making it reusable